policy-service/src/main/java/uk/gov/gchq/palisade/service/policy/response/AuditMessage.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.policy.response;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.util.Assert;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.rule.Rules;
+import uk.gov.gchq.palisade.service.policy.response.common.domain.User;
+
+
+/**
+ * Audit information for a request provided by each service.
+ * Each individual service sends a record to the Audit Service for every request that it receives.
+ * The components of the message will differ depending on which service has sent the data and if the processing was
+ * successful or not.
+ *
+ */
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class AuditMessage {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public final String timeStamp; //when the service processed the request
+    public final String serverIp;   //server for the service that processed the request
+    public final String serverHostname;  //server host name for the service
+
+    public final Context context;  //Context of the client's request
+
+    public final User user;        //User  for the client.  Can be null if there is a userId
+
+    public final LeafResource resource; //Resource for the client.  Can be null if there is a Resource Id
+
+    public final Rules rules;   //Rules that apply to the request.  Can be null if this has not been generated yet.
+
+    public final String errorMessage;  //Error message that occurred during thge processing of the request.  Will be null if there was no issue.
+
+    @SuppressWarnings("checkstyle:parameterNumber")
+    @JsonCreator
+    private AuditMessage(
+            final @JsonProperty("timeStamp") String timeStamp,
+            final @JsonProperty("serverIp") String serverIp,
+            final @JsonProperty("serverHostname") String serverHostname,
+            final @JsonProperty("context") Context context,
+            final @JsonProperty("user") User user,
+            final @JsonProperty("resource") LeafResource resource,
+            final @JsonProperty("rules") Rules rules,
+            final @JsonProperty("errorMessage") String errorMessage) {
```

Comment:
Probably best to not start duplicating these auditmessages everywhere yet, keep them just in palisade/data/audit services